server/src/immich/api-v1/asset/asset.service.ts

```diff
@@ -280,7 +280,13 @@ export class AssetService {
     const asset = await this._assetRepository.getById(assetId);
 
     if (allowExif) {
-      return mapAsset(asset);
+      const data = mapAsset(asset);
+
+      if (data.people) {
+        data.people = data.people.filter((person) => !person.isHidden);
+      }
```

Comment:
Can you move this into the mapper in asset response dto

Reply:
like this : 
```ts
people: entity.faces?.map(mapFace).filter((person) => !person.isHidden),
```

Reply:
Yeah, looks good.